src/components/Question/Question.js

```diff
@@ -109,26 +119,70 @@ class Question extends Component {
     });
   }
 
+  handleKeyDown(event) {
+    /* don't trigger navigation if the current url isn't a question */
+    const pathname = this.props.history.location.pathname;
+    if (!pathname.match(/\/ch\d\/q\d/)) return false;
+
+    /* assign navigation variables */
+    const ARROW_LEFT = 'ArrowLeft';
+    const ARROW_RIGHT = 'ArrowRight';
+    const { previous, next } = this.state.navigation;
+
+    /* handle navigation */
+    switch (event.key) {
+      case ARROW_LEFT:
+        event.preventDefault();
+        previous.enabled
+          ? this.props.history.push(previous.url)
+          : window.addEventListener(
+              'keydown',
+              event => this.handleKeyDown(event),
+              { once: true }
+            );
+        break;
+      case ARROW_RIGHT:
+        event.preventDefault();
+        next.enabled
+          ? this.props.history.push(next.url)
+          : window.addEventListener(
+              'keydown',
+              event => this.handleKeyDown(event),
+              { once: true }
+            );
+        break;
+      default:
+        window.addEventListener('keydown', event => this.handleKeyDown(event), {
+          once: true,
+        });
+        break;
+    }
+  }
+
   toggleExplanationRequest() {
     this.setState({
       explanationRequested: !this.state.explanationRequested,
     });
   }
+  componentDidMount() {
+    /**
+     * register a keydown listener on the window that propagates exactly once
+     * to prevent unnecessary key repeats
+     * have to re-register the listener if this.props.history.push() is not fired
+     * see handleKeyDown()
+     */
+    window.addEventListener('keydown', event => this.handleKeyDown(event), {
+      once: true,
+    });
+  }
+  componentWillUnmount() {
+    window.removeEventListener('keydown', event => this.handleKeyDown(event));
```

Comment:
This won't actually remove the event listener because it doesn't know that this anonymous function is the same as the other anonymous function. Maybe it would be better to hard bind `handleKeyDown` to `Question` class in the `constructor`:

`this.handleKeyDown = this.handleKeyDown.bind(this)`

And then make it the callback:

`window.addEventListener('keydown', this.handleKeyDown);`

And

`window.removeEventListener('keydown', this.handleKeyDown);`

Reply:
Agree!